code/game/machinery/jukebox.dm

```diff
@@ -84,26 +174,41 @@
 		to_chat(usr, SPAN_WARNING("You must secure \the [src] first."))
 		return
 
-	if(stat & (NOPOWER|BROKEN))
+	if(inoperable())
 		to_chat(usr, "\The [src] doesn't appear to function.")
 		return
 
 	if(href_list["change_track"])
-		for(var/datum/track/T in tracks)
-			if(T.title == href_list["title"])
-				current_track = T
-				start_playing()
-				break
+		var/datum/track/T = locate(href_list["change_track"]) in tracks
+		if(istype(T))
+			current_track = T
+			StartPlaying()
+	else if(href_list["loopmode"])
+		var/newval = text2num(href_list["loopmode"])
+		loop_mode = sanitize_inlist(newval, list(JUKEMODE_NEXT, JUKEMODE_RANDOM, JUKEMODE_REPEAT_SONG, JUKEMODE_PLAY_ONCE), loop_mode)
+	else if(href_list["volume"])
+		var/newval = input("Choose Jukebox volume (0-100%)", "Jukebox volume", round(volume * 100.0))
+		newval = sanitize_integer(text2num(newval), min = 0, max = 100, default = volume * 100.0)
+		volume = newval / 100.0
+		update_music() // To broadcast volume change without restarting song
 	else if(href_list["stop"])
-		stop_playing()
+		StopPlaying()
 	else if(href_list["play"])
 		if(emagged)
 			playsound(src.loc, 'sound/items/AirHorn.ogg', 100, 1)
 			for(var/mob/living/carbon/M in ohearers(6, src))
+				var/ear_safety = 0
```

Comment:
For the future: ear safety should get it's own proc instead of copying code every time